src/components/App.vue

```diff
@@ -0,0 +1,35 @@
+<template>
+  <div id="app">
+    <template v-if="$store.state.user">
+      <h2>You are logged in!</h2>
+      <div class="logout">
+        <button @click="logout">Logout</button>
+      </div>
+      <router-view></router-view>
+    </template>
+    <template v-else>
+      <div class="login">
+        <Auth />
+      </div>
+    </template>
+  </div>
+</template>
+
+<script>
+import Auth from './Auth';
+
+export default {
+  name: 'app',
+  components: {
+    Auth,
+  },
+  methods: {
+    logout() {
+      this.$store.dispatch('LOGOUT');
+    },
+  },
+  mounted() {
```

Comment:
mounted is for stuff that happens at a particular stage during page load (fairly early on)

Reply:
It's not about page load. It's about the life cycle of this particular component. So it will happen when this component is put in the page. But not when that component is updated.

If the component is removed and later put back in, it will run again.